mlflow/server/js/src/components/CompareRunContour.js

```diff
@@ -0,0 +1,305 @@
+import './CompareRunContour.css';
+import CompareRunUtil from './CompareRunUtil';
+
+class CompareRunContour extends Component {
+  static propTypes = {
+    runInfos: PropTypes.arrayOf(RunInfo).isRequired,
+    metricLists: PropTypes.arrayOf(Array).isRequired,
+    paramLists: PropTypes.arrayOf(Array).isRequired,
+    runDisplayNames: PropTypes.arrayOf(String).isRequired,
+  };
+
+  // Size limits for displaying keys and values in our plot axes and tooltips
+  static MAX_PLOT_KEY_LENGTH = 40;
+  static MAX_PLOT_VALUE_LENGTH = 60;
+
+  constructor(props) {
+    super(props);
+
+    this.entities = new AllHtmlEntities();
+
+    this.metricKeys = CompareRunUtil.getKeys(this.props.metricLists, false);
+    this.paramKeys = CompareRunUtil.getKeys(this.props.paramLists, false);
+
+    if (this.paramKeys.length + this.metricKeys.length < 3) {
+      this.state = {disabled: true};
+    } else {
+      this.state = {
+        disabled: false,
+        reverseColor: false,
+        xaxis: this.paramKeys.length > 0 ?
+        {
+          key: this.paramKeys[0],
+          isMetric: false,
+        } : {
+          key: this.metricKeys[1],
+          isMetric: true,
+        },
+
+        yaxis: this.paramKeys.length > 0 ?
+        {
+          key: this.paramKeys[1],
+          isMetric: false,
+        } : {
+          key: this.metricKeys[1],
+          isMetric: true,
+        },
+
+        zaxis: this.metricKeys.length > 0 ?
+        {
+          key: this.metricKeys[0],
+          isMetric: true,
+        } : {
+          key: this.paramKeys[1],
+          isMetric: false,
+        },
+      };
+    }
+  }
+
+  /**
+   * Get the value of the metric/param described by {key, isMetric}, in run i
+   */
+  getValue(i, {key, isMetric}) {
+    const value = CompareRunUtil.findInList(
+      (isMetric ? this.props.metricLists : this.props.paramLists)[i], key);
+    return value === undefined ? value : value.value;
+  }
+
+  /**
+   * Encode HTML entities in a string (since Plotly's tooltips take HTML)
+   */
+  encodeHtml(str) {
+    return this.entities.encode(str);
+  }
+
+  getColorscale() {
+    /*
+    * contour plot has an option named "reversescale" which
+    * reverses the color mapping if True, but it doesn't work properly now.
+    *
+    * https://github.com/plotly/plotly.js/issues/4430
+    *
+    * This function is a temporary workaround for the issue.
+    */
+    const colorscale = [
+      [0, 'rgb(5,10,172)'],
+      [0.35, 'rgb(40,60,190)'],
+      [0.5, 'rgb(70,100,245)'],
+      [0.6, 'rgb(90,120,245)'],
+      [0.7, 'rgb(106,137,247)'],
+      [1, 'rgb(220,220,220)'],
+    ];
+
+    if (this.state.reverseColor) {
+      return colorscale;
+    } else {
+      // reverse only RGB values
+      return colorscale.map(([val], index) => [
+        val,
+        colorscale[colorscale.length - 1 - index][1],
+      ]);
+    }
+  }
+
+  render() {
+    if (this.state.disabled) {
+      return <div/>;
```

Comment:
Maybe
```suggestion
      return <div>Contour plots can only be rendered when comparing a group of runs with three or more unique metrics or params. Log more metrics or params to your runs to visualize them using the contour plot.</div>;
```

suggestion, so that users know why the plot is empty